### PR TITLE
Support defaulted values in satfunc data

### PR DESCRIPTION
**Pull request by berland**
Solves #339

### CODE REVIEW

File: ecl2df/common.py
Comment (oyvindeide) [inline ecl2df/common.py:300]: Use enumerate?

File: ecl2df/common.py
Comment (oyvindeide) [inline ecl2df/common.py:301]: Is double underscore intended and needed here?

File: tests/test_satfunc.py
Comment (oyvindeide) [inline tests/test_satfunc.py:313]: This should probably be split into different tests.

File: ecl2df/satfunc.py
Comment (oyvindeide) [inline ecl2df/satfunc.py:125]: Should perhaps have a unit test?

File: ecl2df/common.py
Comment (berland) [reply ecl2df/common.py:301]: It is intended, I have not found another way to do it through `opm`s api.

File: ecl2df/common.py
Comment (berland) [reply ecl2df/common.py:300]: Would like to, but I get: `TypeError: 'opm.libopmcommon_python.DeckItem' object is not iterable`

File: tests/test_satfunc.py
Comment (berland) [reply tests/test_satfunc.py:313]: fixed.

File: ecl2df/satfunc.py
Comment (berland) [reply ecl2df/satfunc.py:125]: yes. will make!

File: tests/test_satfunc.py
Comment (oyvindeide) [inline tests/test_satfunc.py:539]: I think these are good, but personally I have started using `pytest.param` and `id=` which I think is great for readability when using `parameterize`:
```suggestion
            pytest.param(pd.DataFrame(
                [
                    {"SW": 0, "FOO": 0, "SATNUM": 1},
                    {"SW": 0.1, "SATNUM": 1},
                    {"SW": 1, "FOO": 1, "SATNUM": 1},
                ],
            pd.DataFrame(
                [
                    {"SW": 0, "FOO": 0, "SATNUM": 1},
                    {"SW": 0.1, "FOO": 0.1, "SATNUM": 1},
                    {"SW": 1, "FOO": 1, "SATNUM": 1},
                ]
       id="Will not care about column names",
        ),
```
I think the syntax is valid, but did not check it.

Comment (berland) [reply tests/test_satfunc.py:539]: Agree, this looks good. Will fix.

File: tests/test_satfunc.py
Comment (oyvindeide) [inline tests/test_satfunc.py:604]: I am a bit torn on this, I like the compactness. Personally I would put valid/invalid in two separate functions, but will leave it up to you.

File: ecl2df/common.py
Comment (oyvindeide) [reply ecl2df/common.py:301]: Is the `__defaulted` a part of the published api?

File: tests/test_satfunc.py
Comment (berland) [reply tests/test_satfunc.py:604]: Just separate out `xfail` to e.g. `test_interpolate_defaults_xfails()` or also to rewrite the expected failures to something that "passes" pytest?

Comment (oyvindeide) [reply tests/test_satfunc.py:604]: > also to rewrite the expected failures to something that "passes" pytest?

That is my personal preference, for example:
```
@pytest.mark.parametrize(
    "error, error_msg",
    [
        (ValueError, "something"),
        (TypeError, "something"),
    ],
)
def test_my_func(error, error_msg):
    with pytest.raises(error, match=error_msg):
        raise ValueError("something")
``` 
but again, it is my personal preference, so up to you.

Comment (berland) [reply tests/test_satfunc.py:604]: I think I will skip that for now, as it would imply also rewriting other existing test functions.

File: ecl2df/common.py
Comment (berland) [reply ecl2df/common.py:301]: Not quite sure, but have not gotten any hint to other and better alternatives in https://github.com/OPM/opm-common/issues/2598

Comment (oyvindeide) [reply ecl2df/common.py:301]: Ok, then I think that information would be good to add in a comment and/or the commit body.

Comment (berland) [reply ecl2df/common.py:306]: Added in comment.

Comment (oyvindeide) [reply ecl2df/common.py:306]: Technically not private but name mangling (though they could have intended it as private :shrug:)

**Review by oyvindeide — APPROVED** [review]
Dont think I have write access, but I approve!

**Pull request merged**
